### PR TITLE
Remove no longer used elements from AppRepoList 

### PR DESCRIPTION
**Pull request by antgamdia**
### Description of the change

This PR is one of the "repos API revamp" series. In this one, we simply remove what is no longer used in the upcoming move toward the new API.
More details inline.

Note this PR is targeting a temporary branch for the UI, so merging it won't make it be included in main's.

### Benefits

No actual feature-wise benefits, but less code to review later on in follow-up PRs.

### Possible drawbacks

We lose the "force resync" feature, but it is no longer available in the new Repos API.

### Applicable issues

- related #4764

### Additional information

image

### CODE REVIEW

File: dashboard/src/components/Config/AppRepoList/AppRepoButton.tsx [hunk 1]
```diff
@@ -48,9 +51,9 @@ export function AppRepoAddButton({
     ociRepositories: string[],
     skipTLS: boolean,
     passCredentials: boolean,
-    filter?: IAppRepositoryFilter,
+    filter?: IPkgRepositoryFilter,
   ) => {
-    if (repo) {
+    if (packageRepoRef) {
```
Comment (antgamdia) [inline dashboard/src/components/Config/AppRepoList/AppRepoButton.tsx:56]: Just a couple of renames

File: dashboard/src/components/Config/AppRepoList/AppRepoList.tsx [hunk 2]
```diff
@@ -94,27 +95,56 @@ function AppRepoList() {
...
+      : namespacedRepos.push(repo);
   });
 
   const tableColumns = [
     { accessor: "name", Header: "Name" },
     { accessor: "url", Header: "URL" },
+    { accessor: "plugin", Header: "Package Type" },
     { accessor: "accessLevel", Header: "Access Level" },
     { accessor: "namespace", Header: "Namespace" },
+    { accessor: "status", Header: "Status" },
```
Comment (antgamdia) [inline dashboard/src/components/Config/AppRepoList/AppRepoList.tsx:111]: Adding more columns to the table but with hardcoded values for now.

File: dashboard/src/components/Config/AppRepoList/AppRepoList.tsx [hunk 3]
```diff
@@ -94,27 +95,56 @@ function AppRepoList() {
...
         name: getRepoNameLinkAndTooltip(cluster, repo),
         url: repo.spec?.url,
-        accessLevel: repo.spec?.auth?.header ? "Private" : "Public",
+        // TODO(agamez): the PackageRepositorySummary API doesn't expose this field. It will be added in upcoming PRs; in the meantime, set to "unknown"
+        // accessLevel: repo.type?.auth?.header ? "Private" : "Public",
+        accessLevel: "unknown",
         namespace: repo.metadata.namespace,
+        plugin: "Helm",
+        // eslint-disable-next-line no-constant-condition
+        status: true ? (
```
Comment (antgamdia) [inline dashboard/src/components/Config/AppRepoList/AppRepoList.tsx:125]: Leaving this change here, but will be modified and tested in the next PRs

File: dashboard/src/components/Config/AppRepoList/AppRepoList.tsx [hunk 4]
```diff
@@ -155,80 +184,83 @@ function AppRepoList() {
...
-                ) : (
-                  <p>
-                    There are no <i>global</i> Package Repositories yet. Click on the "Add Package
-                    Repository" button to create one.
-                  </p>
-                )}
-                {namespace !== globalReposNamespace && (
-                  <>
-                    <h3>Namespaced Repositories: {namespace}</h3>
+      <div className="catalog-container">
```
Comment (antgamdia) [inline dashboard/src/components/Config/AppRepoList/AppRepoList.tsx:187]: Fixing a css issue: leaving more space at the bottom.

image

File: dashboard/src/shared/url.ts [hunk 5]
```diff
@@ -130,9 +131,6 @@ export const api = {
   // access the k8s api server directly).
   k8s: {
     base: (cluster: string) => `api/clusters/${cluster}`,
-    namespaces: (cluster: string) => `${api.k8s.base(cluster)}/api/v1/namespaces`,
-    namespace: (cluster: string, namespace: string) =>
-      namespace ? `${api.k8s.namespaces(cluster)}/${namespace}` : `${api.k8s.base(cluster)}/api/v1`,
```
Comment (antgamdia) [inline dashboard/src/shared/url.ts:135]: Unused, but looking forward to removing kubeops and operators to clean up the whole file 😝

File: dashboard/src/shared/utils.test.ts [hunk 6]
```diff
@@ -0,0 +1,158 @@
+// Copyright 2022 the Kubeapps contributors.
```
Comment (antgamdia) [inline dashboard/src/shared/utils.test.ts:1]: Adding a test for the whole utils.ts file as we didn't have any yet.

File: dashboard/src/shared/utils.ts [hunk 7]
```diff
@@ -137,6 +136,32 @@ export function getPluginPackageName(plugin?: Plugin | PluginNames | string) {
   }
 }
 
+// FIXME(agamez): replace with a proper call to the plugins server (see getPluginsServiceClientImpl)
```
Comment (antgamdia) [inline dashboard/src/shared/utils.ts:139]: This is a workaround for setting the `Plugin` object later on. However, we still need a way to get (from the UI) the current plugins (closely related to #3889 )

File: dashboard/src/components/Config/AppRepoList/AppRepoList.tsx (hunk 3 above)
Comment (absoludity) [reply dashboard/src/components/Config/AppRepoList/AppRepoList.tsx:125]: Sometimes what I do in this situation, where I'm not targetting the main branch, is that I actually write the test to fail so I can't forget to fix it in the follow-up branch.

File: dashboard/src/components/Config/AppRepoList/AppRepoList.tsx (hunk 4 above)
Comment (absoludity) [reply dashboard/src/components/Config/AppRepoList/AppRepoList.tsx:187]: Great, thanks.

File: dashboard/src/shared/utils.test.ts (hunk 6 above)
Comment (absoludity) [reply dashboard/src/shared/utils.test.ts:1]: Excellent, that'll encourage more tests :)

**Review by absoludity — APPROVED** [review]

**Pull request merged**